test/integration/pack/test_pack.py

```diff
@@ -1732,9 +1733,291 @@ def test_rocks_noncache_flag(cartridge_cmd, light_project, tmpdir, pack_format):
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert "Using cached path .rocks" not in output
+    assert "Using cached path first-dir" not in output
+    assert "Using cached path second-dir/nested" not in output
+
+    cache_dir_items = os.listdir(project_path_cache)
+    assert len(cache_dir_items) == 3
+
+    nested_dir_items = os.listdir(os.path.join(project_path_cache, "second-dir/nested"))
+    assert len(nested_dir_items) == 1
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert "Using cached path .rocks" in output
+    assert "Using cached path first-dir" in output
+    assert "Using cached path second-dir/nested" in output
+
+    new_cache_dir_items = os.listdir(project_path_cache)
+    assert len(new_cache_dir_items) == 3
+    assert cache_dir_items == new_cache_dir_items
+
+
+@pytest.mark.parametrize('pack_format', ['tgz'])
+def test_invalid_yml_params(cartridge_cmd, light_project, tmpdir, pack_format):
+    project = light_project
+
+    # Combine always-cache and key params
+    new_cache_yml_path = os.path.join(tmpdir, "new-pack-cache.yml")
+    with open(new_cache_yml_path, "w") as f:
+        yaml.dump([{"path": ".rocks", "always-cache": True, "key": "just-key"}], f)
+
+    replace_project_file(project, "pack-cache-config.yml", new_cache_yml_path)
+    if os.path.exists(get_rocks_cache_path()):
+        shutil.rmtree(get_rocks_cache_path())
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Please, specify one and only one of `always-true`, `key` and `key-path` for path .rocks" in output
+    assert len(os.listdir(get_rocks_cache_path())) == 0
+
+    # Combine key-path and key params
+    with open(new_cache_yml_path, "w") as f:
+        yaml.dump([{
+            "path": ".rocks",
+            "key-path": os.path.basename(get_rockspec_path(project.path, project.name, "scm-1")),
+            "key": "just-key",
+        }], f)
+
+    replace_project_file(project, "pack-cache-config.yml", new_cache_yml_path)
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Please, specify one and only one of `always-true`, `key` and `key-path` for path .rocks" in output
+    assert len(os.listdir(get_rocks_cache_path())) == 0
+
+    # Invalid key-path file (non exists)
+    with open(new_cache_yml_path, "w") as f:
+        yaml.dump([{"path": ".rocks", "key-path": "invalid_path"}], f)
+
+    replace_project_file(project, "pack-cache-config.yml", new_cache_yml_path)
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Failed to get specified file for path .rocks" in output
```

Comment:
This error looks strange. 

`Failed to get specified cache key file for .rocks`?